### PR TITLE
Avoid multiple rounding of intervals for polynomials

### PR DESCRIPTION
**Pull request by Gbury**
Another instance of multiple rounding causing unsoundness. In essence the fix is very similar to what was done in PR #249 to fix Issue #248 .

### CODE REVIEW

File: src/lib/reasoners/intervalCalculus.ml [hunk 1]
```diff
@@ -675,17 +682,21 @@ let mult_bornes_vars vars env ty =
...
-
+  (* in the case of integer polynomes, ensure that we only round once
+     at the end, and not at every add/scale operation on intervals. *)
+  let rational_interval =
+    List.fold_left
+      (fun i (a, x) ->
+         let i_x, _  =
+           try MX.n_find x env.monomes
+           with Not_found ->
+             if monomes_inited then assert false;
```
Comment (Stevendeo) [inline src/lib/reasoners/intervalCalculus.ml]: Style : could be simplified by `assert (not monomes_inited)`

File: src/lib/reasoners/intervals.ml [hunk 2]
```diff
@@ -73,7 +73,7 @@ module Debug = struct
       print_interval fmt e;
       List.iter (fprintf fmt " U %a" print_interval) l
 
-  let print fmt { ints; expl = e; _ } =
+  let print fmt { ints; expl = e; is_int = _; } =
```
Comment (Stevendeo) [inline src/lib/reasoners/intervals.ml]: Style : is this change necessary?

File: src/lib/reasoners/intervals.ml [hunk 3]
```diff
@@ -492,29 +492,34 @@ let scale_interval_neg n (b1, b2) =
 
 let affine_scale ~const ~coef uints =
   Options.tool_req 4 "TR-Arith-Axiomes scale";
-  if Q.equal coef Q.one then
-    { uints with ints = List.map (translate const) uints.ints; }
```
Comment (Stevendeo) [inline src/lib/reasoners/intervals.ml:496]: Why did you remove the case where the coef is one?

File: src/lib/reasoners/polynome.ml [hunk 4]
```diff
@@ -168,7 +168,7 @@ module Make (X : S) = struct
       else begin
         M.iter
           (fun t n -> fprintf fmt "%s*%a " (Q.to_string n) X.print t) p.m;
-        fprintf fmt "%s" (Q.to_string p.c);
+        fprintf fmt "+ %s" (Q.to_string p.c);
```
Comment (Stevendeo) [inline src/lib/reasoners/polynome.ml]: Style : if `p.c < 0`, you may want not to print `+`

File: src/lib/reasoners/intervals.ml (hunk 3 above)
Comment (Gbury) [reply src/lib/reasoners/intervals.ml:496]: Because it was actually incorrect: if the constant for the translation is rational, then integer intervals need to be rounded even when the coefficient for the scaling is `1`.

Additionally, even if the constant for the translation is an integer, it is useful to use this function as a "round this interval" function, which is precisely what the new `coerce` function does (it performs an `affine_scale` of `coef=1` and `const=0`).

File: src/lib/reasoners/polynome.ml (hunk 4 above)
Comment (Gbury) [reply src/lib/reasoners/polynome.ml]: fixed

File: src/lib/reasoners/intervalCalculus.ml (hunk 1 above)
Comment (Gbury) [reply src/lib/reasoners/intervalCalculus.ml]: done

File: src/lib/reasoners/intervals.ml (hunk 2 above)
Comment (Gbury) [reply src/lib/reasoners/intervals.ml]: not necessary, I reverted the change

**Comment by Gbury** [comment]
I pushed a commit to address the review.

**Comment by Stevendeo** [comment]
> I pushed a commit to address the review.

Thanks! 
Now the CI complains because of the indentation but after this it is fine for me
